instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/AbstractMetricsTest.java

```diff
@@ -7,60 +7,77 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-import io.opentelemetry.sdk.OpenTelemetrySdk;
 import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.data.DoublePoint;
+import io.opentelemetry.sdk.metrics.data.DoubleSummaryPoint;
+import io.opentelemetry.sdk.metrics.data.LongPoint;
 import io.opentelemetry.sdk.metrics.data.MetricData;
-import io.opentelemetry.sdk.metrics.data.MetricData.DoublePoint;
-import io.opentelemetry.sdk.metrics.data.MetricData.LongPoint;
-import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import io.opentelemetry.sdk.metrics.data.MetricDataType;
+import io.opentelemetry.sdk.metrics.data.Point;
 import io.opentelemetry.sdk.metrics.export.IntervalMetricReader;
 import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 
-public class AbstractMetricsTest {
+class AbstractMetricsTest {
   TestMetricExporter testMetricExporter;
 
+  static SdkMeterProvider meterProvider;
+
+  @BeforeAll
+  static void initializeOpenTelemetry() {
+    meterProvider = SdkMeterProvider.builder().buildAndRegisterGlobal();
+  }
+
   @BeforeEach
-  public void beforeEach() {
+  void beforeEach() {
     testMetricExporter = new TestMetricExporter();
   }
 
   IntervalMetricReader createIntervalMetricReader() {
     return IntervalMetricReader.builder()
         .setExportIntervalMillis(100)
         .setMetricExporter(testMetricExporter)
-        .setMetricProducers(
-            Collections.singletonList(
-                OpenTelemetrySdk.getGlobalMeterProvider().getMetricProducer()))
+        .setMetricProducers(Collections.singletonList(meterProvider))
         .build();
   }
 
   public void verify(
-      String metricName, String unit, MetricData.Type type, boolean checkNonZeroValue) {
+      String metricName, String unit, MetricDataType type, boolean checkNonZeroValue) {
     List<MetricData> metricDataList = testMetricExporter.metricDataList;
     for (MetricData metricData : metricDataList) {
       if (metricData.getName().equals(metricName)) {
         assertThat(metricData.getDescription()).isNotEmpty();
         assertThat(metricData.getUnit()).isEqualTo(unit);
-        assertThat(metricData.getPoints()).isNotEmpty();
+        metricData.getDoubleGaugeData().getPoints();
+        List<Point> points = new ArrayList<>();
```

Comment:
@jkwatson Don't know if it matters, but the impact of losing `getPoints` was relatively large. Presumably I should not be instanceof below, but without doing a deeper dive into this test it's not easy to know what's a long gauge or a long sum so used this pattern.

Reply:
paging @bogdandrutu

Reply:
This doesn't look like a huge impact to me. Am I missing something?

Reply:
I would actually merge this logic with the next 20 lines:

switch metric.getType() {
  case GaugeDouble:
    List<DoublePoints> dpts = metricData.getDoubleGaugeData().getPoints();
    dpts.foreach(doublePoint -> assertThat(doublePoint.getValue()).isGreaterThan(0.0));
  ....
}

Or something like that (written from the phone so no access to IDE 👍 )

Reply:
I would have to understand the test much better to e.g. understand whether to use gauge vs sum. Will go with my brain-dead change for now and hopefully someone can give some maintenance to the test, maybe after the metrics stuff stabilizes some more.